spec.md

```diff
@@ -0,0 +1,238 @@
+|<img src="https://github.com/sylabs/singularity/blob/master/docs/logos/singularity_v3.png?raw=true" width=30x>|[Singularity][singularity], by [Sylabs][sylabs]|`application/vnd.sylabs.sif.config.v1+json`|
+
+## Defining a Unique Artifact Type
+
+Defining the unique type involves uniqueness for computer processing, and uniqueness for humans.
+
+For computer processing, artifacts are defined by setting `manifest.config.mediaType` to a globally unique value.
+
+> **Note:** The `config.mediaType` of `application/vnd.oci.image.config.v1+json` is reserved for artifacts intended to be instanced by [docker][docker] and [containerd][containerd].  
+*Each artifact MUST have its own unique type.*
+
+ The following `mediaType` format is used to differentiate the type of artifact:
+
+`application/vnd.`[org|company]`.`[objectType]`.`[optionalSubType]`.config.`[version]`+json`
+
+- **`org|company`** - represents an open source foundation (`oci`, `cncf`) or a company (`microsoft`, `ibm`).
+- **`objectType`** - a value representing the short name of the type. The combination of the org|company and object type should provide human identification and uniqueness for computers indexing.
+- **`optionalSubType`** - provides additional extensibility of an `objectType`
+- **`version`** - provides artifact authors the ability to revision their schemas and formatting, enabling tools to identify which format they will process.
+
+## Defining Supported Layer Types
+
+Artifacts are intended to have content. The content of an artifact is represented through a collection of blobs as provided by the [distribution spec][distribution-spec]. How the blobs are re-constituted, and whether the blobs are ordinal layers is a decision of the artifact author.
+
+As an example, [OCI Images][image-layer] are represented through an ordinal collection of compressed files. Each blob represents a layer. Each layer overlays the previous layer.
+
+Other artifacts may be represented as a single file, such as config artifact representing a deployment specification. Other artifacts may include a config blob, and a collection of binaries compressed as another blob. By separating the blob, the artifact author can benefit from layer de-duplication and concurrent downloading of each blob.
+
+### Layer File Format
+
+Layers are persisted as blobs within registries. The blobs can be a single file or a collection of files. The persistance format is up to the artifact author. They may choose to persist individual files with their native or custom format, such as `.config`, `.json`, `.bin`, `.sif`, and compress them with various formats such as `.gzip`.
+
+Large files can benefit from compression when being transferred across the network. However, decompression takes time and compute cycles. For smaller, individual files, the decompression may take longer than downloading the file in its original format.
+
```

Comment:
It might be worth adding that transport compression and storage compression can be used instead of compressed blobs; this is arguably a better model longer term, although currently registries may not support these.

Reply:
I'm moving this to the [artifact authors section](https://github.com/opencontainers/artifacts/pull/6) as it's more in the opinion category.